### ISSUE TITLE
Questions about Bladesinger

### ISSUE THREAD

**Issue by ALFAIthildur** [AbdelAdrian/Eldritch_Magic#2]
The Bladesinger kit is a Fighter/Mage kit in 2e. The implementation of Argent77's QDMulti mod enables much more flexibility with multiclass kits than previously possible - why the choice to make Bladesingers bards of all things in this mod? Add to that the fact this component completely morphs  default bards into something unrecognizable and incompatible with many established bard mods like Rogue Rebalancing... Please reconsider making Bladesinger a proper fighter/mage kit as it correctly should be, and leave default bards as they are.

**Comment by AbdelAdrian** [comment]
To be honest, I'm baffled at your comment. First of all, please read the readme. As with many of my kits, there are multiple options. The bladesinger is the perfect example, being my oldest and most polished kit. I wrote that long before Aquadrizzt's QDMulti mod, and as such, bard was a workaround. Still long before Aquadrizzt's QDMulti mod, I used in-game items/NPCs/scripts to apply kits to multiclass characters in-game after character creation, and thus the Fighter/Mage variant of the bladesinger was born. I don't want to say definitively that I was the first to adopt Aquadrizzt's QDMulti mod when he published it, but I was one of the first to adopt Aquadrizzt's QDMulti mod when he published it - this mod is even listed on the OP of that thread, and I'm tagged in post 2 as well: https://forums.beamdog.com/discussion/59743/tool-qdmulti-a-library-for-multiclass-kits/p1

I can forgive not reading the readme, or knowing I was heavily involved with using the QDMulti mod after its publication, but the reason I'm baffled is that all your questions, comments, and concerns would be assuaged in the installation process in which you get to pick various types of the same kit, especially useful if you're running pre-2.0 and a true kitted MC is impossible in chargen, which leads me to believe you have not installed, much less used, any component of Eldritch Magic.

Hopefully I'm not coming across as a big jerk whose answered the same questions for years, but ultimately it is my fault for continuing to support pre-2.0 MC kits and it has been my intention for a very long time to heavily redact my mod and limit confusion. I don't know when I'll release the next version (soon™), but going forward there will be one bladesinger and it will be a fighter/mage made in chargen - anyone wanting "obsolete" bard or in-game hackery versions will be free to download old versions from archive.

In the meantime, I genuinely hope you still have some interest in my mods, and if so I really do recommend reading the readme, going through the installation process slowly, and selecting the post-2.0 kits, which are by far the best of the lot anyway. If you don't use pre-2.0 kits, none of the core classes will be affected, leaving default bards exactly as you want them.

**Comment by ALFAIthildur** [comment]
I apologize, but I did read the readme... the information I found and read must've been from an older source (can't remember exactly where, might've been a BD forum post). I'll look for the updated version with the kit as a proper fighter/mage. Kudos!

**Comment by ALFAIthildur** [comment]
Should I disregard this in the readme file? I'm seeing this right after the description for Bladesinger, in the readme file from version 1.68, which appears to be the latest version.

_Changes to all Bards: Elven Bards enabled, Fighter APR & proficiencies (first level & rate),
Fighter max proficiences for first & other levels, bards may now wear helmets and robes,
& use fighter-only potions. Bard spell progression becomes that of mages while XP/level
 reflects that of a fighter/mage multiclass._
Definitely not something I'd want for my install.


Under class features:
-Can you explain d7 HD rather than the normal Fighter/mage HD progression?
-How is 'proficient in blind fighting' implemented?
-The RP restriction listed under class features: 'Must attempt to advance the cause of elvendom somehow at all times, and lend aid to any elf in need' - the fact it's listed under class features rather than with the subsequent CBoE description makes me wonder - is this is somehow implemented into the kit?
-'May cast arcane spells one-handed with a +2 casting time penalty.' The IE games obviously don't requires 1 OR 2 hands to be free for casting. Is this actually implemented? An actual penalty would make little sense given that no spellcasting class in the game is penalized whether they have one, two, or zero hands free (i.e. in this game 10APR dual wielding fighter/mages cast without any issue whatsoever with zero hands free; it would be odd to implement a penalty for someone that has one hand free).
-What is Divine Intervention/Seldarine's Chosen and how do they work?

Thanks for your time; this does look like the best version of the Bladesinger kit I've seen so far (and I've seen several).

**Comment by AbdelAdrian** [comment]
Yeah, I double-checked the readme; it does explain, at least to the best of my ability, the multiple choices and relevant instructions for each. The GitHub formatting doesn't help, but perhaps you didn't use the scroll bar, as this redacted quote is from the very same line of text "Bladesinger...(Fighter/Mage or Bard kit)*" 
If, and only if, the pre-2.0 bard version of the bladesinger is installed, then the core bard class will also be modified. The recommended post-2.0 F/M version will not make any changes to bards, nor will any other component of this mod. So yeah, disregard, next update will remove all of this backwards compatibility, and hopefully, confusion.

- d7 HD means up to 7 HP per level.
- immune to blindness
- RP is just RP, I try to make this very clear as well. I also don't hide the fact that the description is largely from the 2E kit. Bladesingers are expected to act a certain way, almost like chaotic elven paladins. If you want to make a bladesinger and go around murdering every elf, by all means, but it was my intent for people to RP as a champion of the elven people. It's up to you though.
- Yes, this is actually implemented and is the coolest friggen thing ever imo. Credit to subtledoctor.
Yeah, yeah, yeah, I know it doesn't make sense in an IE game. A F/M can dual-wield and cast spells at normal speed, so why are bladesingers slower casters even with an extra hand free? Because this is based on the 2nd Ed. AD&D Complete Book of Elves - if it was in the description AND doable in the IE, I did it (with a lot of help, like with weapons & armor affecting casting). If it was in the description but I couldn't manage it, I tried to make this clear while still leaving the original class description as much as possible.
- They're virtually unattainable Godhood level secret rewards that are simply granted to you upon your undying loyalty to leveling your bladesinger. They're cheesy, and stupid, and ridiculously pointless at the level received, but they're fun in my opinion.

Thank you for your vote of confidence, enjoy!

**Comment by ALFAIthildur** [comment]
It looks like the blindness immunity is applied multiple times for some reason if I'm reading things correctly: at lvl 1 from EMFM.spl, then levels 7 and 8 from EMSEE.spl . Is this intentional/is there something I'm overlooking?

Re: the casting penalty,
> this is actually implemented and is the coolest friggen thing ever imo
> **I know it doesn't make sense in an IE game**
> if it was in the description AND doable in the IE, I did it

That's an equation that doesn't add up in my view; however it's easily tweakable via NI, so no biggie.
AC bonus seems closer to IE Swashbuckler (-1 every 5 levels) than CBoE Bladesinger, but that too can be tweaked a bit; probably a headache to implement the situational AC bonus exactly per CBoE. 
The  option to put 5 pips into Single Weapon style for Bladesingers is also nonexistent in the book (or any 2e supplement AFAIK). 
The immunity to Blindness is, obviously, not the same as Blind fighting, and might be considered overpowered.

Bottom line, it's pretty obvious most of the PnP features do not translate exactly into IE, making the insistence on casting penalty somewhat odd. If anything, logic would say always having a hand free like the Bladesinger should _**facilitate**_ casting compared to having both hands occupied like virtually every caster in these games (dual wielding F/M, sword and shield Clerics, mages with staff, etc all cast without problems). It actually makes more sense to reverse the speed penalty to a bonus if anything (bonus of 2 is probably too much however); thematically it fits Elves (speed, fluidity, agility) like the mod's bonus to weapon speed. Interestingly, 3rd edition Bladesinger's Song of Celerity expands on a similar theme, casting swiftly and attacking.

I'm not too concerned about the kit being overpowered presently (testing the tweaks in current playthrough); having to have offhand free is a tremendous sacrifice for a BG2/ToB F/M due to the boatloads of heavily enchanted weapons and shields in this game, many with permanent status effects that are incredibly valuable or even life saving (Belm, Crom Feyr, Harmony shield, Shield of Balduran, etc etc), and of course, APR is king as far as melee characters are concerned in this game (loss of up to 4 APR from offhand Belm/Kundane w/ Imp Haste or HLA). I think this sacrifice brings some balance; essentially Bladesinger is probably powerful for BG1 levels but eventually gets overshadowed by a dual wield F/M in mid/late BG2 and ToB.

I'll definitely be doing some tweaking, but thanks for the work you've put in to make this. It's good stuff, in spite of some odd decisions it's still the closest offering to a 2e Bladesinger I've seen so far.

**Comment by ALFAIthildur** [comment]
One more note: I started noticing that I was getting multiple castings of each variety of Bladesong available in the abilities menu matching the number of times I activated the ability (got up to 17 castings of Defensive Bladesong before I noticed - eventually I believe this can cause some problems if the number gets too high). 
Each of the bladesong.spl files need a Remove Spell added to counteract the Give Innate Ability function so that this doesn't happen each time the ability is activated; it's a common convention for innate abilities that are not Vancian, i.e. not limited to x/day, in the IE engine. I'm guessing you are aware of this already and it was just an oversight.
Cheers.
